clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -329,17 +329,18 @@ private TransactionalRequestResult beginCompletingTransaction(TransactionResult
     }
 
     public synchronized TransactionalRequestResult sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offsets,
-                                                                            String consumerGroupId) {
+                                                                            final ConsumerGroupMetadata groupMetadata,
+                                                                            final boolean enableGroupFencing) {
         ensureTransactional();
         maybeFailWithError();
         if (currentState != State.IN_TRANSACTION)
             throw new KafkaException("Cannot send offsets to transaction either because the producer is not in an " +
                     "active transaction");
 
-        log.debug("Begin adding offsets {} for consumer group {} to transaction", offsets, consumerGroupId);
+        log.debug("Begin adding offsets {} for consumer group {} to transaction", offsets, groupMetadata);
```

Comment:
`ConsumerGroupMetadata` does not have a proper `toString()` implementation -- if we want to log the object, we should add `ConsumerGroupMetadata#toString()` to ensure a readable log message.